ui/components/app/transaction-activity-empty-state/transaction-activity-empty-state.tsx

```diff
@@ -1,37 +1,83 @@
-import React from 'react';
+import React, { useCallback } from 'react';
+import { useSelector } from 'react-redux';
 import { twMerge } from '@metamask/design-system-react';
+import { InternalAccount } from '@metamask/keyring-internal-api';
+import { EthMethod } from '@metamask/keyring-api';
 import { TabEmptyState } from '../../ui/tab-empty-state';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import { useTheme } from '../../../hooks/useTheme';
+import { getUseExternalServices, getIsSwapsChain } from '../../../selectors';
+import { getCurrentChainId } from '../../../../shared/modules/selectors/networks';
+import { MetaMetricsSwapsEventSource } from '../../../../shared/constants/metametrics';
+import useBridging from '../../../hooks/bridge/useBridging';
 import { ThemeType } from '../../../../shared/constants/preferences';
+import { getMultichainNetwork } from '../../../selectors/multichain';
+import { useMultichainSelector } from '../../../hooks/useMultichainSelector';
+import { MultichainNetworks } from '../../../../shared/constants/multichain/networks';
 
 export type TransactionActivityEmptyStateProps = {
   /**
    * Additional className to apply to the component
    */
   className?: string;
+  /**
+   * The account to use for swap logic
+   */
+  account: InternalAccount;
 };
 
 export const TransactionActivityEmptyState: React.FC<
   TransactionActivityEmptyStateProps
-> = ({ className }) => {
+> = ({ className, account }) => {
   const t = useI18nContext();
   const theme = useTheme();
+  const isSigningEnabled =
+    account.methods.includes(EthMethod.SignTransaction) ||
+    account.methods.includes(EthMethod.SignUserOperation);
+  const isExternalServicesEnabled = useSelector(getUseExternalServices);
+  const chainId = useSelector(getCurrentChainId);
+  const isSwapsChain = useSelector((state) => getIsSwapsChain(state, chainId));
+
+  // Get multichain network info for Solana detection (same as coin-buttons)
+  const { chainId: multichainChainId } = useMultichainSelector(
+    getMultichainNetwork,
+    account,
+  );
```

Comment:
Noticed that when solana network was selected the swap button was disabled so using this approach from coin-buttons.tsx as well